src/node_debug_options.h

```diff
@@ -0,0 +1,52 @@
+#ifndef SRC_NODE_DEBUG_OPTIONS_H_
+#define SRC_NODE_DEBUG_OPTIONS_H_
+
+#include <string>
+
+// Forward declaration to break recursive dependency chain with src/env.h.
+namespace node {
+
+enum class DebugAgentType {
+  kNone,
+  kDebugger,
+#if HAVE_INSPECTOR
+  kInspector
+#endif  // HAVE_INSPECTOR
+};
+
+class DebugOptions {
+ public:
+  DebugOptions();
+  bool ParseOption(const std::string& option);
+  DebugAgentType EnabledAgentType() const {
+#if HAVE_INSPECTOR
+    if (inspector_enabled_) {
+      return DebugAgentType::kInspector;
+    }
+#endif  // HAVE_INSPECTOR
+    if (debug_enabled_) {
+      return DebugAgentType::kDebugger;
+    }
+    return DebugAgentType::kNone;
+  }
```

Comment:
Can you add `bool IsInspector() const` convenience methods?  That would let you simplify conditionals in other files.